File: fixtures/go-multimodule/services/billing/pkg/invoice/invoice.go
package invoice

import (
	"strconv"
	"strings"
)

type Service struct {
	MaxRefundCents int
}

func ParseInvoiceTotal(input string) int {
	value := strings.TrimSpace(strings.TrimPrefix(input, "total="))
	total, err := strconv.Atoi(value)
	if err != nil {
		return 0
	}
	return total
}

func ApplyDiscountCents(cents int, percent int) int {
	if percent < 0 {
		percent = 0
	}
	if percent > 100 {
		percent = 100
	}
	return cents - cents*percent/100
}

func NormalizeToken(input string) string {
	token := strings.TrimSpace(input)
	if token == "" {
		return "anonymous"
	}
	return strings.ToLower(token)
}

func (s Service) AuthorizeRefund(role string, amount int) bool {
	return role == "admin" || (role == "support" && amount <= s.MaxRefundCents)
}

File: fixtures/sample-typescript/src/invoice.ts
export type Role = "admin" | "support" | "viewer";

export function parseInvoiceTotal(raw: string): number | null {
  const trimmed = raw.trim();
  if (trimmed.length === 0) {
    return null;
  }

  const cents = Number.parseInt(trimmed, 10);
  if (!Number.isSafeInteger(cents) || cents < 0) {
    return null;
  }
  return cents;
}

export function normalizeToken(token?: string): string {
  return (token ?? "").trim().toLowerCase();
}

export class RefundPolicy {
  authorizeRefund(role: Role, cents: number): boolean {
    if (cents <= 0 || cents > 50_000) {
      return false;
    }
    return role === "admin" || (role === "support" && cents <= 5_000);
  }
}
